safehttp/form.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"errors"
+	"mime/multipart"
+	"strconv"
+)
+
+// Form TODO
+type Form struct {
+	values map[string][]string
+	err    error
+	parsed bool
+}
+
+// MultipartForm TODO
+type MultipartForm struct {
+	form Form
+	file map[string][]*multipart.FileHeader
+}
+
+// Int TODO
+func (f *Form) Int(paramName string, defaultValue int) int {
+	if f.err != nil {
+		return defaultValue
+	}
+	if !f.parsed {
+		panic("form has not been parsed")
+	}
+	vals, ok := f.values[paramName]
+	if !ok {
+		f.err = errors.New("no value found for key " + paramName)
```

Comment:
```suggestion
		f.err = fmt.Printf("no value found for key %q", paramName)
```

Reply:
I'm adding this mostly for `%q` that adds quotes around the string and does escaping as needed.

Reply:
```
x := f.Int("x", 0)
y := f.Int("y", 0)
if err := f.Error(); err != nil {
  return err // here the information about which one errored is lost
}

Reply:
Resolved in [2dc01f6](url)